### PR TITLE
[Feature:TAGrading] Allow user to remove hotkey

### LINKED ISSUE #10621: Toggle/Remove TA Hotkeys
**What problem are you trying to solve with Submitty**

TA Grading hotkeys can be rebound, but they cannot be turned off entirely. Some TAs have expressed that they would prefer to remove hotkeys from their interface.

**Describe the way you'd like to solve this problem**

When a hotkey is rebound, there should also be an option to unbind it from any key.

### PR DESCRIPTION
**Pull request by ZheyuDeng**
Fix #10621 
### What is the current behavior?
Users were only able to remove hotkey back to default key instead of completly remove it. It also gets pretty unusable.
image

### What is the new behavior?
User are now able to remove shrotcut. Also added buttons to restore all and remove all hotkeys.

https://github.com/Submitty/Submitty/assets/122962727/fba0846f-0726-4078-8801-3c7ae6fc3236

### Other information?

When testing you might need to clear browser cache.
Since some keys are locked from changing should user be able to remove them? If so how should they restore.

### CODE REVIEW

**Review by sahilsuman933 — CHANGES_REQUESTED** [review]
It works properly, but it would be nice to have a button through which we can get the default settings.﻿

https://github.com/Submitty/Submitty/assets/34382211/a934f525-6dc4-49a6-ab3a-34bb95ad7222

**Comment by sahilsuman933** [comment]
@ZheyuDeng js-lint is failing can you please fix it?

**Comment by manzil-infinity180** [comment]
adding cypress test for **restore all** and **remove all** button could be better to test

**Comment by ZheyuDeng** [comment]
I am currently writing cypress test for it a review for functionality/UI will be great!

**Review by zacharymnp — CHANGES_REQUESTED** [review]
Functionality is good. Two small comments:

1. Every button was unset when I first opened the page. I had to click Restore All. The keybinds should probably be enabled by default.
2. I would recommend 'Unassigned' or 'Unbound' instead of 'Unset'.

**Review by zacharymnp — APPROVED** [review]

File: site/cypress/e2e/Cypress-Gradeable/hotkey.spec.js [hunk 1]
```diff
@@ -47,4 +48,21 @@ describe('TA grading hotkey testing', () => {
...
         cy.get('[data-testid="posts-list"]').should('contain', 'Discussion Posts');
     });
+
+    it('allows user to remove hotkey', () => {
+        cy.login();
+        cy.visit(['sample', 'gradeable', 'grading_homework', 'grading', 'details']);
+        cy.get('[data-testid="view-sections"]').click();
+        cy.get('[data-testid="grade-button"]').eq(12).click();
+        cy.get('[data-testid="grading-setting-btn"]').click();
+        cy.get('#remap-unset-0').click();
```
Comment (IDzyre) [inline site/cypress/e2e/Cypress-Gradeable/hotkey.spec.js:58]: Can you create testids for the remap elements here?

**Review by IDzyre — CHANGES_REQUESTED** [review]

File: site/cypress/e2e/Cypress-Gradeable/hotkey.spec.js [hunk 2]
```diff
@@ -47,4 +48,21 @@ describe('TA grading hotkey testing', () => {
...
+        cy.get('[data-testid="view-sections"]').click();
+        cy.get('[data-testid="grade-button"]').eq(12).click();
+        cy.get('[data-testid="grading-setting-btn"]').click();
+        cy.get('[data-testid="remap-unset-0"]').click();
+        cy.get('[data-testid="remap-0"]').should('contain', 'Unassigned');
+        cy.get('[data-testid="remove-all-hotkeys"]').click();
+        cy.get('[data-testid="remap-1"]').should('contain', 'Unassigned');
+        cy.get('[data-testid="remap-2"]').should('contain', 'Unassigned');
+        cy.get('[data-testid="restore-all-hotkeys"]').click();
+        cy.get('[data-testid="remap-0"]').should('not.contain', 'Unassigned');
```
Comment (IDzyre) [inline site/cypress/e2e/Cypress-Gradeable/hotkey.spec.js:64]: Is there a value that you should expect to be here, not just `not unassigned`?

Comment (ZheyuDeng) [reply site/cypress/e2e/Cypress-Gradeable/hotkey.spec.js]: Ye your right that makes a lot of sense

**Review by IDzyre — APPROVED** [review]
Tests look good, didn't test more than that.

**Review by manzil-infinity180 — APPROVED** [review]
Tested locally , looks good to me

**Review by sahilsuman933 — APPROVED** [review]
LGTM 🚀

**Pull request merged**